src/vdaf.rs

```diff
@@ -250,6 +252,25 @@ pub trait Aggregator<const VERIFY_KEY_SIZE: usize, const NONCE_SIZE: usize>: Vda
     ) -> Result<Self::AggregateShare, VdafError>;
 }
 
+/// Aggregator that implements differential privacy with Aggregator-side noise addition.
+#[cfg(feature = "experimental")]
+pub trait AggregatorWithNoise<
+    const VERIFY_KEY_SIZE: usize,
+    const NONCE_SIZE: usize,
+    DPStrategy: DifferentialPrivacyStrategy,
```

Comment:
How is a VDAF meant to indicate compatibility with an implementation of `DifferentialPrivacyStrategy`? Is the expectation that `add_noise_to_agg_share` returns an error if the DP strategy is not compatible?

Reply:
With the current design we determine compatibility at compile time. I think we can indicate compatibility for a given VDAF by implementing something like this:
```rust
impl AggregatorWithNoise<16, 16, ZCdpDiscreteGaussian> for Prio3SumVec {
	fn add_noise_to_agg_share(
		&self,
		dp_strategy: &ZCdpDiscreteGaussian,
		agg_param: &Self::AggregationParam,
		agg_share: &mut Self::AggregateShare,
		num_measurements: usize,
		) -> Result<(), VdafError> {
			// Compute the L2 sensitivity (because we are using Gaussian noise)
			// Call `dp_strategy.create_distribution` (because the budget is in zCDP form)
			// Add i.i.d. noise
			todo!();
		}
}
```

For instance, we won't be able to compile a DAP program that passes a `dp_strategy` of type `EpsilonDeltaLaplace` to a `Prio3SumVec` VDAF, unless there is a corresponding implementation. Would that work?

Reply:
Yes, I understand how this works now. Thanks!